docs/key-concepts.md

```diff
@@ -0,0 +1,122 @@
+# Key Concepts
+
+Observability and monitoring are crucial in modern software development and operations. It’s all about understanding, monitoring, and fixing complex systems using data collection, analysis, and visualization. When organizations use observability and monitoring effectively, they can spot and solve problems quickly, improve performance, and ensure their applications and infrastructure run smoothly.
+
+Observability and monitoring essentially mean keeping a close eye on how systems behave and perform using logs and metrics. These logs and metrics are generated by various components within the system, including applications, servers, databases, and networking devices. They serve as valuable sources of information for understanding system health, diagnosing issues, and optimizing performance.
+
+In this module, we’ll explore the ***key concepts of observability*** and understand how logs and metrics play a crucial role in providing insights into system behaviour and performance.
+
+## 1. Logs 📝
+
+Logs are simply the **records of events** or actions that occur within a system. Each log entry typically contains relevant information such as **timestamps**, **event descriptions**, **severity levels**, and **contextual data**. Logs serve as a detailed history of system activity, providing valuable insights into system behaviour, errors, and performance issues.
+
+These logs are essential for troubleshooting problems, auditing system activity, and understanding user interactions. In software development and operations, logs are generated by various components, including applications, servers, databases, and networking devices.
+
+This is what a sample log data in a system would look like.
+
+| Timestamp            | Event Description                       | Severity | Contextual Data                                    |
+| -------------------- | --------------------------------------- | -------- | -------------------------------------------------- |
+| 2024-02-11T08:35:00Z | Application server started successfully | INFO     | Server: app-01, Environment: Production            |
+| 2024-02-11T09:12:45Z | Disk usage exceeded threshold           | WARNING  | Server: db-03, Disk Usage: 85%, Threshold: 80%     |
+| 2024-02-11T10:20:30Z | Critical error in database connection   | ERROR    | Server: db-02, Error Code: 500, Connection Timeout |
+| 2024-02-11T11:05:15Z | User authentication failed              | ERROR    | Username: johndoe, IP: 192.168.1.100               |
+| 2024-02-11T12:30:00Z | Scheduled job completed successfully    | INFO     | Job: Backup, Duration: 00:45:00                    |
+
```

Comment:
instead of writing these manually, would we want to attach a screenshot of some Log data?

Reply:
Even I thought the same. Apparently, the images don't look sharp enough, So I'll probably see if I can get a more intuitive image for this log data.